beacon-chain/p2p/interfaces.go

```diff
@@ -59,6 +58,12 @@ type ConnectionHandler interface {
 	connmgr.ConnectionGater
 }
 
+// SenderEncoder allows sending functionality from libp2p as well as encoding for requests and responses.
```

Comment:
Smaller interface surface area instead of depending on the entire `P2P` interface